third_party/chromium/media/filters/source_buffer_state.cc

```diff
@@ -915,9 +915,43 @@ bool SourceBufferState::OnNewConfigs(
   return success;
 }
 
+#if defined(STARBOARD)
+void SourceBufferState::SetSourceBufferStreamMemoryLimit(size_t limit) {
+  stream_memory_limit_override_ = limit;
+  SetStreamMemoryLimits();
+}
+
+size_t SourceBufferState::GetSourceBufferStreamMemoryLimit() {
+  // a source buffer can be backed my multiple Demuxer streams, although at
+  // YouTube we usually only have a single one. If we've set an override then
+  // all values will be the same, but we shouldn't assume that, so instead we'll
+  // return the largest here if there are multiple.
+  std::vector<size_t> memory_limits;
+  for (const auto& it : audio_streams_) {
+    memory_limits.push_back(it.second->GetStreamMemoryLimit());
+  }
+  for (const auto& it : video_streams_) {
+    memory_limits.push_back(it.second->GetStreamMemoryLimit());
+  }
+
+  if (memory_limits.empty()) return 0;
+
+  return *std::max_element(memory_limits.begin(), memory_limits.end());
+}
```

Comment:
size_t memory_limit = 0;
for (..) {
  memory_limit = std::max(memory_limit, ...);
}
return memory_limit;

Reply:
nice!